projects/kit/components/input-phone/input-phone.component.ts

```diff
@@ -166,56 +156,41 @@ export class TuiInputPhoneComponent
     }
 
     get canClean(): boolean {
-        return this.computedValue !== this.countryCode && this.textfieldCleaner.cleaner;
-    }
-
-    onDrop(event: DragEvent): void {
-        if (!event.dataTransfer) {
-            return;
-        }
-
-        this.setValueWithoutPrefix(event.dataTransfer.getData('text'));
-        event.preventDefault();
+        return (
+            this.nativeValue !== this.nonRemovablePrefix && this.textfieldCleaner.cleaner
+        );
     }
 
-    onPaste(event: Event): void {
-        this.setValueWithoutPrefix(tuiGetClipboardDataText(event as ClipboardEvent));
+    get maskOptions(): MaskitoOptions {
+        return this.calculateMask(
+            this.countryCode,
+            this.phoneMaskAfterCountryCode,
+            this.nonRemovablePrefix,
+            this.allowText,
+        );
     }
 
     onActiveZone(active: boolean): void {
         this.updateFocused(active);
 
-        if (active && !this.computedValue && !this.readOnly && !this.allowText) {
-            this.updateSearch(this.countryCode);
+        if (active && !this.nativeValue && !this.readOnly && !this.allowText) {
+            this.updateSearch(this.nonRemovablePrefix);
+            this.nativeValue = this.nonRemovablePrefix;
 
             return;
         }
 
-        if (
-            this.computedValue === this.countryCode ||
-            (this.search !== null &&
-                Number.isNaN(
-                    parseInt(this.search.replace(TUI_MASK_SYMBOLS_REGEXP, ''), 10),
-                ))
-        ) {
-            this.updateSearch('');
+        if (!active && !this.allowText && this.nativeFocusableElement) {
+            this.nativeValue = this.nativeValue.replace(/-$/, '');
```

Comment:
Why only hyphen is removed? Can it be space?

Reply:
I can change to this way (not digit):

```suggestion
            this.nativeValue = this.nativeValue.replace(/\D$/, '');
```

What do you think about it ?

Reply:
LGTM